cordapp-example/workflows-kotlin/src/main/kotlin/com/example/utils/MoneyFinder.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.utils
+
+import com.example.schema.CashSchemaV1
+import com.example.state.CashState
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.identity.Party
+import net.corda.core.node.ServiceHub
+import net.corda.core.node.services.Vault
+import net.corda.core.node.services.vault.PageSpecification
+import net.corda.core.node.services.vault.QueryCriteria
+import net.corda.core.node.services.vault.builder
+
+// just for this exercise and make it simple `ExampleFlow` creates a money, so on payback I can use
+// "the same" money (same owner, same value). This class finds it.
+object MoneyFinder {
+    fun call(serviceHub: ServiceHub, cashOwner: Party, amount: Int): StateAndRef<CashState> {
+        val pageSpec = PageSpecification(pageNumber = 0, pageSize = 1)
+        val criteria = moneyQueryCriteria(cashOwner, amount.toLong())
+        val results = serviceHub.vaultService.queryBy(CashState::class.java, criteria = criteria, paging = pageSpec)
+
+        return results.states.first()
```

Comment:
Of course, if the user has 2 Cash states with 2 different amounts, who knows which state will be first.